packages/core-utils/src/query.js

```diff
@@ -174,21 +210,48 @@ function parseLocationString(value) {
     : null;
 }
 
+/**
+ * Convert a string query param for a from or to place into a location. If
+ * coordinates not provided and geocoder config is present, use the first
+ * geocoded result.
+ * @param  {string} value
+ * @param  {Object} [geocoderConfig=null]
+ * @return {Location}
+ */
+async function queryParamToLocation(value, geocoderConfig = null) {
+  let location = parseLocationString(value);
+  if (!location && value && geocoderConfig) {
+    // If a valid location was not found, but the place name text exists,
+    // attempt to geocode the name.
+    location = await getFirstGeocodeResult(value, geocoderConfig);
+  }
+  return location;
+}
+
 /**
  * Create a otp query based on a the url params.
  *
  * @param  {Object} params An object representing the parsed querystring of url
  *    params.
+ * @param config the config in the otp-rr store.
  */
-export function planParamsToQuery(params) {
+export async function planParamsToQuery(params, config) {
   const query = {};
-  Object.keys(params).forEach(key => {
+  // FIXME? Must use for to execute the awaits statements one after the other
+  // and obtain a populated query object.
+  /* eslint-disable no-await-in-loop */
```

Comment:
Of all these disablers, I think this `no-await-in-loop` matters the most here and will affect app performance since making a geocode request for both the from and to locations will happen sequentially instead of asynchronously. If we're looking to publish this change fast, then we can ignore this for now, but if we're ever looking at performance this will definitely be a hotspot that needs eventual correction.